Detector/DetSegmentation/DetSegmentation/GridDriftChamber.h

```diff
@@ -28,62 +29,96 @@ class GridDriftChamber : public Segmentation {
   virtual Vector3D position(const CellID& aCellID) const;
   virtual CellID cellID(const Vector3D& aLocalPosition, const Vector3D& aGlobalPosition,
                         const VolumeID& aVolumeID) const;
+  virtual double distanceTrackWire(const CellID& cID, const TVector3& hit_start, const TVector3& hit_end) const;
 
-//  inline double innerRadius() const { return m_innerRadius; }
-//  inline double detectorLength() const { return m_detectorLength; }
-  inline double offsetPhi() const { return m_offsetPhi; }
-  inline double delta_phi() const{ return m_delta_phi; }
+  inline double cell_Size() const { return m_cellSize; }
+  inline double offset_phi() const { return m_offsetPhi; }
+  inline double epsilon0() const { return m_epsilon0; }
+  inline double detectorLength() const { return m_detectorLength; }
   inline const std::string& fieldNamePhi() const { return m_phiID; }
   // Setters
 
-//  inline void setGeomParams(int layer, double sizePhi) {
-//    layer_params[layer] = {sizePhi};
-// }
-
-//  void updateParams(int layer) const {
-//    auto it_end = layer_params.cend();
-//    --it_end;
-//    double size = it_end->second[0];
-//    double radius = it_end->second[1];
-//    double eps = it_end->second[2];
-//
-//    auto map_it = layer_params.find(layer);
-//    if (map_it != layer_params.cend()) {
-//      size = map_it->second[0];
-//      radius = map_it->second[1];
-//      eps = map_it->second[2];
-//    }
-//
-//    _currentGridSizePhi = size;
-//    _currentRadius = radius;
-//    m_epsilon = eps;
-//  }
-
   inline double phiFromXY(const Vector3D& aposition) const {
     return std::atan2(aposition.Y, aposition.X) + M_PI ;
   }
 
-//  inline int returnLayer(double x, double y) const {
-//  // Hit R position
-//    double R = std::sqrt(x * x + y * y);
-//  // Layer
-//    int layer = int((R - m_innerRadius) / m_cellSize);
-//    return layer;
-//  }
+  inline void setGeomParams(int layer, double layerphi, double R, double eps ) {
+     layer_params[layer] = {layerphi,R,eps};
+   }
+
+  inline void setWiresInLayer(int layer, int numWires)
+  {
+    double phi0;
+    updateParams(layer);
+    for (int i = 0; i<numWires; ++i) {
+
+    if(layer % 2 == 0) { phi0 = 0.; }
+    else { phi0 = 0.5 * _currentLayerphi; }
```

Comment:
Hi @myliu-hub ,
Do you think that it would better to move the initialization of phi0 to the detector construction or the constructor of this class?
Tao